src/mlpack/core/tree/spill_tree/spill_tree_impl.hpp

```diff
@@ -0,0 +1,753 @@
+/**
+ * @file spill_tree_impl.hpp
+ *
+ * Implementation of generalized hybrid spill tree (SpillTree).
+ */
+#ifndef MLPACK_CORE_TREE_SPILL_TREE_SPILL_TREE_IMPL_HPP
+#define MLPACK_CORE_TREE_SPILL_TREE_SPILL_TREE_IMPL_HPP
+
+// In case it wasn't included already for some reason.
+#include "spill_tree.hpp"
+
+namespace mlpack {
+namespace tree {
+
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename HyperplaneMetricType> class HyperplaneType,
+         template<typename SplitMetricType, typename SplitMatType>
+             class SplitType>
+SpillTree<MetricType, StatisticType, MatType, HyperplaneType, SplitType>::
+SpillTree(
+    const MatType& data,
+    const double tau,
+    const size_t maxLeafSize,
+    const double rho) :
+    left(NULL),
+    right(NULL),
+    parent(NULL),
+    count(0),
+    pointsIndex(NULL),
+    overlappingNode(false),
+    hyperplane(),
+    bound(data.n_rows),
+    parentDistance(0), // Parent distance for the root is 0: it has no parent.
+    dataset(&data),
+    localDataset(false)
+{
+  arma::Col<size_t> points;
+  if (dataset->n_cols > 0)
+    // Fill points with all possible indexes: 0 .. (dataset->n_cols - 1).
+    points = arma::linspace<arma::Col<size_t>>(0, dataset->n_cols - 1,
+        dataset->n_cols);
```

Comment:
Maybe just using `regspace()` is the better thing to do here, but I'm indifferent either way.